File: Code/C/Inputs.c
#include <stdio.h>
#include <string.h>

int main() {
    printf("Hello, World! (From C)\n");

    char msg[10];

    while (true) {
        printf("Say Hi: ");
        fgets(msg, sizeof(msg), stdin);
        msg[strcspn(msg, "\n")] = 0;

        if (strcmp(msg, "Hi") == 0) {
            printf("Hello!\n");
        } else if (strcmp(msg, "Quit") == 0) {
            printf("See you later!\n");
            break;
        } else {
            printf("Inncorrect\n");
        }
    }

    return 0;
}

File: Code/Haskell/Inputs.hs
import qualified Data.Map as Map
import Data.Char (toLower)

main :: IO ()
main = do
    putStrLn "Hello, World! (From haskell)"
    loop

responses :: Map.Map String String
responses = Map.fromList [("hi", "Hello!"), ("bye", "Why are you leaving?")]

loop :: IO ()
loop = do
    putStr "Say Hi: "
    msg <- getLine
    let msgLower = map toLower msg

    if msgLower == "quit" then
        putStrLn "See you later!"
    else
        case Map.lookup msgLower responses of
            Just reply -> do
                putStrLn reply
                loop
            Nothing -> do
                putStrLn "I don't understand what you mean."
                loop

